src/wp-includes/blocks.php

```diff
@@ -856,15 +856,19 @@ function get_comment_delimited_block_content( $block_name, $block_attributes, $b
  * @return string String of rendered HTML.
  */
 function serialize_block( $block, $callback = null ) {
-	if ( is_callable( $callback ) ) {
-		$block = call_user_func( $callback, $block );
-	}
-
 	$block_content = '';
 
 	$index = 0;
 	foreach ( $block['innerContent'] as $chunk ) {
-		$block_content .= is_string( $chunk ) ? $chunk : serialize_block( $block['innerBlocks'][ $index++ ], $callback );
+		if ( is_string( $chunk ) ) {
+			$block_content .= $chunk;
+		} else {
+			$inner_block = $block['innerBlocks'][ $index++ ];
+			if ( is_callable( $callback ) ) {
+				$inner_block = call_user_func( $callback, $inner_block, $block );
```

Comment:
To optimize #5240, we could even pass the block (and/or chunk) index:

```suggestion
				$inner_block = call_user_func( $callback, $inner_block, $block, $index );
```

Reply:
In this scenario, `$block` is the `$inner_block`'s parent block?

Reply:
Yeah, exactly! See [L866](https://github.com/WordPress/wordpress-develop/pull/5242/files#diff-8c99af92e4ec0fdb307ddd9b42be1e1ef1efe4a9f31287c23f346244dddd1ce9R866) 😊 

The nice part here is that we'll have a lot of information available for free: parent block, block index, chunk index. We can pass them to the callback, from which they'll eventually be passed to `insert_hooked_block`. That way, we won't have to [search](https://github.com/WordPress/wordpress-develop/pull/5240/files#diff-8c99af92e4ec0fdb307ddd9b42be1e1ef1efe4a9f31287c23f346244dddd1ce9R840) for the anchor block's position during sibling insertion -- which should be a significant optimization 😄

Reply:
I'll make that change quickly.